zerver/lib/i18n.py

```diff
@@ -67,3 +68,18 @@ def get_and_set_request_language(
     RequestNotes.get_notes(request).set_language = translation.get_language()
 
     return request_language
+
+
+def get_browser_locale(request: HttpRequest) -> Optional[str]:
+    accept_lang_header = request.META.get("HTTP_ACCEPT_LANGUAGE")
+    if accept_lang_header is None:
+        return None
+
+    available_language_codes = get_available_language_codes()
+    for accept_lang, priority in parse_accept_lang_header(request.META.get("HTTP_ACCEPT_LANGUAGE")):
+        if accept_lang == "*":
+            return None
+        if accept_lang not in available_language_codes:
+            continue
+        return accept_lang
```

Comment:
This ensures that we use the language with highest quality value and one that is supported by us.